vue-components/src/components/TransformWidget.vue

```diff
@@ -0,0 +1,56 @@
+<script setup lang="ts">
+import type { ParameterValue, TransformDescription, TransformValue } from '../types'
+
+import ParamsWidget from './ParamsWidget.vue'
+
+interface Props {
+  pos: number
+  value: TransformValue
+  descriptions: { [name: string]: TransformDescription }
+}
+
+type Events = {
+  'update:type': [type: string]
+  'update:params': [params: { [name: string]: ParameterValue }]
+  remove: []
+}
+
+defineProps<Props>()
+const emit = defineEmits<Events>()
+</script>
+
+<template>
+  <div>
+    <q-select
+      filled
+      options-dense
+      stack-label
+      map-options
+      :model-value="value.name"
+      label="Transform"
+      :options="Object.keys(descriptions)"
+      @update:model-value="(t: string) => emit('update:type', t)"
+    >
+      <template v-slot:prepend>
+        <q-chip>{{ pos }}</q-chip>
```

Comment:
Might make sense to put this and the trash button outside the select.

Reply:
That's actually how I had done it initially, but I didn't completely like the fact that it was gonna make the transform select narrower than the input fields of its parameters, see image below. What do you think?

![Screenshot 2024-11-18 at 14-34-30 NRTK Explorer](https://github.com/user-attachments/assets/18dc2dbf-1f15-427a-b462-c82109ea663b)

Reply:
Ah, right, wack a mole. No biggie.